src/frontend/src/handler/alter_table_column.rs

```diff
@@ -286,6 +286,20 @@ pub async fn handle_alter_table_column(
                 ))?
             }
 
+            if new_column
+                .options
+                .iter()
+                .any(|x| matches!(x.option, ColumnOption::NotNull))
+                && !new_column
+                    .options
+                    .iter()
+                    .any(|x| matches!(x.option, ColumnOption::DefaultValue(_)))
+            {
+                return Err(ErrorCode::InvalidInputSyntax(
+                    "alter table add NOT NULL columns must have default value".to_owned(),
```

Comment:
What if the default value is `NULL`? 🤯

Reply:
Hmmm, we should allow this but reject the row violating the NOT NULL constrains, according to PG. However, our current Default expr haven't forbid NULL, which means user can insert `NULL` to NOT NULL column via `default NULL`. We can first keep them aligned.

Reply:
`However, our current Default expr haven't forbid NULL`
I'll leave a TODO to fix it later.